pincer/exceptions.py

```diff
@@ -268,6 +268,10 @@ class TimeoutError(PincerError):
     """
 
 
+class ConnectionError(PincerError):
```

Comment:
maybe GatewayError would be better?

Reply:
I think `GatewayConnectionError` would be best